data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/critical_branch/CriticalBranchReader.java

```diff
@@ -82,99 +82,117 @@ public boolean isSelected() {
 
     public CriticalBranchReader(TBranch tBranch, @Nullable TOutage tOutage, Crac crac, UcteNetworkAnalyzer ucteNetworkAnalyzer, Set<Side> defaultMonitoredSides) {
         String outage;
+        final String basecase = "basecase";
+        boolean isMonitored = false;
         if (tOutage == null) {
-            outage = "basecase";
+            outage = basecase;
         } else {
+            if (tOutage.getV() == null && tOutage.getName() != null) {
+                isMonitored = true;
+                tOutage.setV(tOutage.getName().getV());
+            }
             outage = tOutage.getV();
         }
         this.criticalBranchName = String.join(" - ", tBranch.getName().getV(), tBranch.getFromNode().getV(), tBranch.getToNode().getV(), outage);
-        UcteCnecElementHelper branchHelper = new UcteCnecElementHelper(tBranch.getFromNode().getV(), tBranch.getToNode().getV(), String.valueOf(tBranch.getOrder().getV()), ucteNetworkAnalyzer);
+        UcteFlowElementHelper branchHelper = new UcteFlowElementHelper(tBranch.getFromNode().getV(), tBranch.getToNode().getV(), String.valueOf(tBranch.getOrder().getV()), ucteNetworkAnalyzer);
         this.nativeBranch = new NativeBranch(branchHelper.getOriginalFrom(), branchHelper.getOriginalTo(), branchHelper.getSuffix());
         if (!branchHelper.isValid()) {
             this.isImported = false;
             this.invalidBranchReason = branchHelper.getInvalidReason();
             this.criticalBranchImportStatus = ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK;
             return;
         }
+        if (outage.equals("mneHasTooManyBranches")) {
+            this.isImported = false;
+            this.invalidBranchReason = "MonitoredElement has more than 1 Branch";
+            this.criticalBranchImportStatus = ImportStatus.NOT_YET_HANDLED_BY_FARAO;
+            return;
+        }
         this.monitoredSides = branchHelper.isHalfLine() ? Set.of(Side.fromIidmSide(branchHelper.getHalfLineSide())) : defaultMonitoredSides;
-        if (tOutage != null && crac.getContingency(tOutage.getV()) == null) {
+        if (tOutage != null && crac.getContingency(tOutage.getV()) == null && !tOutage.getV().equals(basecase)) {
             this.isImported = false;
             this.criticalBranchImportStatus = ImportStatus.INCOMPLETE_DATA;
             this.invalidBranchReason = String.format("CNEC is defined on outage %s which is not defined", tOutage.getV());
         } else {
             this.isImported = true;
             this.isDirectionInverted = branchHelper.isInvertedInNetwork();
-            this.selected = isSelected(tBranch);
-            if (tOutage == null) {
+            if (isMonitored) {
+                this.selected = false;
+            } else {
+                this.selected = isSelected(tBranch);
+            }
```

Comment:
replace with this.selected = isMonitored ? false : isSelected(tBranch)
for readability